### ISSUE TITLE
[kubectl] Security vulnerabilites identified in kubectl image

### ISSUE THREAD

**Issue by wong0703** [bitnami/containers#54479]
### Describe your issue as much as you can

Hi

We are currently using the image bitnami/kubectl:1.28.5. As we are in preparation to go live, this image underwent scans by Trivy and Grype, and numerous CVEs were found. May I know is there any plan to remediate the findings (especially for those with medium severity and above)?

Understand VMware Tanzu application catalog may be offering similar image but is there a report of the scan result of that image lying around somewhere so that I may reference and decide? Many thanks!

The following CVEs were reported by Grype and Trivy:

Finding | severity
-- | --
CVE-2019-8457 | CRITICAL
CVE-2023-23914 | CRITICAL
CVE-2023-45853 | CRITICAL
CVE-2020-16156 | HIGH
CVE-2020-22218 | HIGH
CVE-2021-33560 | HIGH
CVE-2022-1304 | HIGH
CVE-2022-3715 | HIGH
CVE-2022-42916 | HIGH
CVE-2022-43551 | HIGH
CVE-2022-4899 | HIGH
CVE-2023-25652 | HIGH
CVE-2023-29007 | HIGH
CVE-2023-2953 | HIGH
CVE-2023-31484 | HIGH
CVE-2023-47038 | HIGH
CVE-2005-2541 | LOW
CVE-2007-5686 | LOW
CVE-2007-6755 | LOW
CVE-2010-0928 | LOW
CVE-2010-4756 | LOW
CVE-2011-3374 | LOW
CVE-2011-3389 | LOW
CVE-2011-4116 | LOW
CVE-2013-0340 | LOW
CVE-2013-4235 | LOW
CVE-2013-4392 | LOW
CVE-2015-3276 | LOW
CVE-2016-2781 | LOW
CVE-2017-11164 | LOW
CVE-2017-14159 | LOW
CVE-2017-16231 | LOW
CVE-2017-17740 | LOW
CVE-2017-18018 | LOW
CVE-2017-7245 | LOW
CVE-2017-7246 | LOW
CVE-2018-1000021 | LOW
CVE-2018-20796 | LOW
CVE-2018-5709 | LOW
CVE-2018-6829 | LOW
CVE-2019-1010022 | LOW
CVE-2019-1010023 | LOW
CVE-2019-1010024 | LOW
CVE-2019-1010025 | LOW
CVE-2019-19882 | LOW
CVE-2019-20838 | LOW
CVE-2019-9192 | LOW
CVE-2020-13529 | LOW
CVE-2020-15719 | LOW
CVE-2021-22922 | LOW
CVE-2021-22923 | LOW
CVE-2021-36084 | LOW
CVE-2021-36085 | LOW
CVE-2021-36086 | LOW
CVE-2021-36087 | LOW
CVE-2022-0563 | LOW
CVE-2022-24975 | LOW
CVE-2022-3219 | LOW
CVE-2022-41409 | LOW
CVE-2022-48303 | LOW
CVE-2023-25815 | LOW
CVE-2023-28320 | LOW
CVE-2023-29383 | LOW
CVE-2023-31437 | LOW
CVE-2023-31438 | LOW
CVE-2023-31439 | LOW
CVE-2023-31486 | LOW
CVE-2023-39804 | LOW
CVE-2023-4016 | LOW
TEMP-0290435-0B57B5 | LOW
TEMP-0517018-A83CE6 | LOW
TEMP-0628843-DBAD28 | LOW
TEMP-0841856-B18BAF | LOW
CVE-2023-23915 | MEDIUM
CVE-2023-4039 | MEDIUM
CVE-2023-46218 | MEDIUM
CVE-2023-46219 | MEDIUM
CVE-2023-4641 | MEDIUM
CVE-2023-4806 | MEDIUM
CVE-2023-4813 | MEDIUM
CVE-2023-50495 | MEDIUM
CVE-2023-5678 | MEDIUM
CVE-2023-5981 | MEDIUM
CVE-2023-48795 | UNKNOWN

**Comment by mpermar** [comment]
@wong0703 , unfortunately there are no reports for the Tanzu Application Catalog images other than for TAC customers, but for your interest, this is what a Trivy outputs for a "VMware photon"-based kubectl container image shows:

```
..../vac_vmware_it/containers/photon-4/kubectl:1.29.0 (photon 4.0)

Total: 7 (UNKNOWN: 0, LOW: 0, MEDIUM: 5, HIGH: 2, CRITICAL: 0)

┌───────────────────────┬───────────────┬──────────┬────────┬───────────────────┬───────────────┐
│        Library        │ Vulnerability │ Severity │ Status │ Installed Version │ Fixed Version │
├───────────────────────┼───────────────┼──────────┼────────┼───────────────────┼───────────────┤
│ openssl-fips-provider │ CVE-2023-4807 │ HIGH     │ fixed  │ 3.0.8-1.ph4       │ 3.0.9-6.ph4   │
│                       │               │          │        │                   │               │
│                       │               │          │        │                   │               │
│                       ├───────────────┤          │        │                   ├───────────────┤
│                       │ CVE-2023-5363 │          │        │                   │ 3.0.9-7.ph4   │
│                       │               │          │        │                   │               │
│                       ├───────────────┼──────────┤        │                   ├───────────────┤
│                       │ CVE-2023-2650 │ MEDIUM   │        │                   │ 3.0.9-1.ph4   │
│                       │               │          │        │                   │               │
│                       ├───────────────┤          │        │                   ├───────────────┤
│                       │ CVE-2023-2975 │          │        │                   │ 3.0.9-2.ph4   │
│                       │               │          │        │                   │               │
│                       │               │          │        │                   │               │
│                       ├───────────────┤          │        │                   ├───────────────┤
│                       │ CVE-2023-3446 │          │        │                   │ 3.0.9-3.ph4   │
│                       │               │          │        │                   │               │
│                       │               │          │        │                   │               │
│                       ├───────────────┤          │        │                   ├───────────────┤
│                       │ CVE-2023-3817 │          │        │                   │ 3.0.9-4.ph4   │
│                       │               │          │        │                   │               │
│                       ├───────────────┤          │        │                   ├───────────────┤
│                       │ CVE-2023-5678 │          │        │                   │ 3.0.9-8.ph4   │
│                       │               │          │        │                   │               │
│                       │               │          │        │                   │               │
└───────────────────────┴───────────────┴──────────┴────────┴───────────────────┴───────────────┘
```

Those 7 CVEs is something the photon team is working on addressing but do not affect the container image. As such, TAC ships with a VEX (CSAF-based) file that can be used for compliance and also for automated filtering.

So, in summary, right TAC provides an effective 0 CVE Kubectl

**Comment by mpermar** [comment]
Fwiw, there are other distributions (Ubuntu, Redhat UBI, ...) that provide different CVE footprint. A product representative can help with evaluating those.

**Comment by wong0703** [comment]
hi @mpermar  @carrodher  thank you for taking your time out to reply my issue. I will probably look at Tanzu Application Catalog images to see if they are better suited for my organization's usage. Appreciate the info!
